### PR TITLE
Add ResourceManager Test

### PR DESCRIPTION
**Pull request by melina5656**
Added testing code in `resource-manager-models-resources.cs`

### CODE REVIEW

File: test/CadlRanchProjects.Tests/resource-manager-models-resources.cs [hunk 1]
```diff
@@ -56,5 +57,163 @@ public class ResourceManagerModelsResourcesTests: CadlRanchTestBase
             Assert.AreEqual("AzureSDK", response.Value.Data.SystemData.LastModifiedBy);
             Assert.AreEqual(CreatedByType.User, response.Value.Data.SystemData.LastModifiedByType);
         });
+
+        [Test]
+        [Ignore("https://github.com/Azure/autorest.csharp/issues/4876")]
+        public Task Azure_ResourceManager_Models_Resources_TopLevelTrackedResources_update() => Test(async (host) =>
+        {
+            var id = TopLevelTrackedResource.CreateResourceIdentifier(Guid.Empty.ToString(), "test-rg", "top");
+            var data = new TopLevelTrackedResourceData()
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/resource-manager-models-resources.cs:66]: Use public constructor

File: test/CadlRanchProjects.Tests/resource-manager-models-resources.cs [hunk 2]
```diff
@@ -56,5 +57,163 @@ public class ResourceManagerModelsResourcesTests: CadlRanchTestBase
...
+            Assert.AreEqual("AzureSDK", response.Value.Data.SystemData.LastModifiedBy);
+            Assert.AreEqual(CreatedByType.User, response.Value.Data.SystemData.LastModifiedByType);
+        });
+
+        [Test]
+        public Task Azure_ResourceManager_Models_Resources_TopLevelTrackedResources_listBySubscription() => Test(async (host) =>
+        {
+            var id = SubscriptionResource.CreateResourceIdentifier(Guid.Empty.ToString());
+            var responses = MgmtTestHelper.CreateArmClientWithMockAuth(host).GetSubscriptionResource(id).GetTopLevelTrackedResourcesAsync();
+            await foreach (var response in responses)
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/resource-manager-models-resources.cs:118]: So there is only one page? Can we assert this?

File: test/CadlRanchProjects.Tests/resource-manager-models-resources.cs [hunk 3]
```diff
@@ -56,5 +57,163 @@ public class ResourceManagerModelsResourcesTests: CadlRanchTestBase
...
+            Assert.AreEqual(CreatedByType.User, response.Value.Data.SystemData.CreatedByType);
+            Assert.AreEqual("AzureSDK", response.Value.Data.SystemData.LastModifiedBy);
+            Assert.AreEqual(CreatedByType.User, response.Value.Data.SystemData.LastModifiedByType);
+        });
+
+        [Test]
+        public Task Azure_ResourceManager_Models_Resources_NestedProxyResources_createOrReplace() => Test(async (host) =>
+        {
+            var id = TopLevelTrackedResource.CreateResourceIdentifier(Guid.Empty.ToString(), "test-rg", "top");
+            var data = new NestedProxyResourceData()
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/resource-manager-models-resources.cs:153]: same as above

File: test/CadlRanchProjects.Tests/resource-manager-models-resources.cs [hunk 4]
```diff
@@ -56,5 +57,163 @@ public class ResourceManagerModelsResourcesTests: CadlRanchTestBase
...
+            Assert.AreEqual(CreatedByType.User, response.Value.Data.SystemData.CreatedByType);
+            Assert.AreEqual("AzureSDK", response.Value.Data.SystemData.LastModifiedBy);
+            Assert.AreEqual(CreatedByType.User, response.Value.Data.SystemData.LastModifiedByType);
+        });
+
+        [Test]
+        public Task Azure_ResourceManager_Models_Resources_NestedProxyResources_update() => Test(async (host) =>
+        {
+            var id = NestedProxyResource.CreateResourceIdentifier(Guid.Empty.ToString(), "test-rg", "top", "nested");
+            var data = new NestedProxyResourceData()
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/resource-manager-models-resources.cs:176]: same as above

**Review by pshao25 — APPROVED** [review]

**Comment by melina5656** [comment]
Link issue: #4843

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**